packages/common/src/tools/LogThrottler.ts

```diff
@@ -0,0 +1,71 @@
+interface LogInfo {
+  count: number
+  throttleCount: number
+  isThrottling: boolean
+}
+
+export class LogThrottler {
+  private readonly recentLogs: Map<string, LogInfo>
+
+  constructor(
+    private readonly threshold: number,
+    private readonly thresholdTimeInMs: number,
+    private readonly throttleTimeInMs: number,
+  ) {
+    this.recentLogs = new Map<string, LogInfo>()
+  }
+
+  add(logKey: string) {
+    let logInfo = this.recentLogs.get(logKey)
+
+    if (!logInfo) {
+      logInfo = {
+        count: 0,
+        isThrottling: false,
+        throttleCount: 0,
+      }
+      this.recentLogs.set(logKey, logInfo)
+    }
+
+    this.incrementCount(logInfo)
+
+    if (logInfo.count >= this.threshold && !logInfo.isThrottling) {
+      this.throttle(logKey, logInfo)
+    }
+  }
+
+  isThrottling(logKey: string) {
+    return this.recentLogs.get(logKey)?.isThrottling
+  }
+
+  private throttle(logKey: string, logInfo: LogInfo) {
+    logInfo.isThrottling = true
+
+    setTimeout(() => {
+      if (logInfo.throttleCount !== 0) {
+        console.log(
+          `[LOG THROTTLER] ${logKey} was logged ${
```

Comment:
we could use "our" logger here but with throttling disabled.

Reply:
I implemented this before and got lint errors
![image](https://user-images.githubusercontent.com/93620601/215564552-0f26f37c-9a12-4519-a3a9-0297929817f7.png)